resources/lib/criterion_parser.py

```diff
@@ -31,7 +31,7 @@ def parse_criterion(criterion):
         filter.update(value)
     elif isinstance(value, list):
         filter['value'] = list(map(lambda v: v['id'], value))
-    else:
+    elif value is not '':
```

Comment:
I think it's better to go for `if 'value' in criterion:`, as an empty string might be a valid filter value.
```suggestion
    elif 'value' in criterion:
```

But it might be that breaks again, not sure. If that's the case the modifier should be checked and at least excluding the `IS_NULL` / `IS_NOT_NULL` modifiers etc as those won't have a value anyway.

Reply:
Everything seems to work for me at least with my set of filters, so with that limited amount of testing I think this should work.